woodwork/accessor_utils.py

```diff
@@ -65,7 +65,7 @@ def _update_column_dtype(series, logical_type):
             series = ks.from_pandas(formatted_series)
         else:
             series = series.apply(_reformat_to_latlong)
-    if logical_type.pandas_dtype != str(series.dtype):
+    if logical_type.pandas_dtype != str(series.dtype):  # --> should this be cheking the valid ltype dtype or just pandas?
```

Comment:
I think we might want to switch to checking if `_get_valid_underlying_dtype(..) != str(series.dtype)` because a series that's using a backup dtype will always go into this code block even if it's the correct dtype